rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/RunControllerV3.scala

```diff
@@ -212,4 +208,13 @@ class RunControllerV3 @Autowired()(runService: RunServiceV3) extends BaseControl
     }
   }
 
+  protected def parseYmdDate(dateOptStr: Optional[String]): Option[LocalDate] = {
+    dateOptStr.toScalaOption.map { dateStr =>
+      Try(LocalDate.parse(dateStr)) match {
+        case Success(parsed) => parsed
+        case Failure(e) => throw new IllegalArgumentException(s"Could not parse YYYY-MM-DD date from $dateStr: ${e.getMessage}")
```

Comment:
The service should not fail, it is just the response that will communicate the problem; with `None`, the invalid date would behave as not given at all.

I would understand this would be preferred if we were changing an implementation already used by users, but since this is V3 API that nobody uses yet, I feel we can afford to be stricter here. But I am willing to be persuaded.

Reply:
Sorry, misunderstood where it's used (not when querying data but when parsing request). This is totally OK of course.